## import tensorflow 函式庫
import進來並確認版本號

In [2]:
import tensorflow as tf
print(tf.__version__)

1.10.0


## 載入Mnist資料集
Load and prepare the [MNIST dataset](http://yann.lecun.com/exdb/mnist/). Convert the samples from integers to floating-point numbers:

In [3]:
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

11493376/11490434 [==============================] - 2s 0us/step


## 建立模型

In [4]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

## 進行訓練

In [5]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test, y_test)


Epoch 1/5
60000/60000 [==============================] - 5s 77us/step - loss: 0.2994 - acc: 0.9129
Epoch 2/5
60000/60000 [==============================] - 4s 73us/step - loss: 0.1447 - acc: 0.9573
Epoch 3/5
60000/60000 [==============================] - 4s 74us/step - loss: 0.1077 - acc: 0.9675
Epoch 4/5
60000/60000 [==============================] - 4s 73us/step - loss: 0.0886 - acc: 0.9728
Epoch 5/5
10000/10000 [==============================] - 0s 23us/step


[0.07879162678141147, 0.9764]

The image classifier is now trained to ~98% accuracy on this dataset. To learn more, read the [TensorFlow tutorials](https://www.tensorflow.org/tutorials/).